### PR TITLE
CPLAT-9778: Update builder readme to reflect new codegen

### PR DESCRIPTION
**Pull request by greglittlefield-wf**
## Motivation
The builder/README.md file wasn't updated when we originally implemented the new parsing/codegen to support the new mixin-based boilerplate.

#### Goals:
- Update the document to reflect the new builder logic

#### Non-Goals
- Revamp that document entirely
- Go through parsing/codegen for  every kind of declaration in detail

## Changes
- Update the document to reflect the new builder logic
- Update generated code snippets
- Update outdated boilerplate in snippets

## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: 

### QA Checklist
- [ ] Markdown renders well
- [ ] Links aren't broken

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] A Client Platform member has reviewed these changes
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_
- [ ] _For release PRs -_ Version metadata in Rosie comment is correct

[contributing-review-types]: https://github.com/Workiva/over_react/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/over_react/blob/master/CONTRIBUTING.md#manual-testing-criteria

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: lib/src/builder/README.md [hunk 1]
```diff
@@ -52,26 +45,74 @@ own components by viewing the generated .over_react.g.dart file.
...
+       
+2. The builder creates a concrete props implementation class, which mixes in the generated props mixin:
+    ```dart
+    class _$$FooProps extends UiProps with FooProps, $FooProps {
+      ...   
+    }
+    ```
+    This concrete class is what will be used everywhere for this component's typed props (returned from the `UiFactory`, available via `props` inside `render`).
+   
+    Note that in the above example, our component has props declared via just a props mixin, which is a shorthand syntax. For components with concrete props classes, the implementation is similar, and mixins in all generated mixins corresponding to the props mixins used:
```
Comment (corwinsheahan-wf) [inline lib/src/builder/README.md:88]: I think this is clearer. Please tweak if it's not 100% accurate tho.
```suggestion
    Note that in the above example, our component has props declared via just a props mixin, which is a shorthand syntax for when no additional props are needed from other mixins. For components with concrete props classes which mix in other props, the implementation is similar. The generated concrete props class mixes in all generated mixins corresponding to the props mixins used:
```

File: lib/src/builder/README.md [hunk 2]
```diff
@@ -52,26 +45,74 @@ own components by viewing the generated .over_react.g.dart file.
...
+    class _$$BarProps extends UiProps 
+        with 
+            BarPropsMixin, $BarPropsMixin, 
+            FooProps, $FooProps
+        implements BarProps {
+      ...   
+    }
+    ``` 
+   
+    It also overrides `UiProps`'s members that were stubbed out, and implements a constructor that delegates to other generated subclasses: one optimized for `JsBackedMap`, and one used for all other `Map`s.
```
Comment (corwinsheahan-wf) [inline lib/src/builder/README.md:106]:
```suggestion
    It also overrides stubbed out `UiProps`'s members, and implements a constructor that delegates to other generated subclasses: one optimized for `JsBackedMap`, and one used for all other `Map`s. 
```

File: lib/src/builder/README.md (hunk 1 above)
Comment (greglittlefield-wf) [reply lib/src/builder/README.md]: Committed!

File: lib/src/builder/README.md (hunk 2 above)
Comment (greglittlefield-wf) [reply lib/src/builder/README.md]: Committed!

**Comment by corwinsheahan-wf** [comment]
QA +1
* CI passes (docs only change)
* All links in changed files work and point to the correct locations.

Merging into master.
@Workiva/release-management-p

**Pull request merged**